src/dune_rules/ml_sources.ml

```diff
@@ -389,29 +389,43 @@ let modules_of_stanzas =
       ; melange_emits = List.rev melange_emits
       }
   in
-  fun stanzas ~expander ~project ~dir ~libs ~lookup_vlib ~modules ~include_subdirs ->
+  fun stanzas
+    ~expander
+    ~project
+    ~dir
+    ~libs
+    ~lookup_vlib
+    ~modules
+    ~include_subdirs
+    ~lib_config ->
     Memo.parallel_map stanzas ~f:(fun stanza ->
       match Stanza.repr stanza with
       | Library.T lib ->
-        (* jeremiedimino: this [Resolve.get] means that if the user writes an
-           invalid [implements] field, we will get an error immediately even if
-           the library is not built. We should change this to carry the
-           [Or_exn.t] a bit longer. *)
-        let+ sources, modules =
-          let lookup_vlib = lookup_vlib ~loc:lib.buildable.loc in
-          make_lib_modules
-            ~expander
-            ~dir
-            ~libs
-            ~lookup_vlib
-            ~modules
-            ~lib
-            ~include_subdirs
-            ~version:lib.dune_version
-          >>= Resolve.read_memo
-        in
-        let obj_dir = Library.obj_dir lib ~dir in
-        `Library (lib, sources, modules, obj_dir)
+        let* lib_config = lib_config in
+        let info = Library.to_lib_info lib ~dir ~lib_config in
+        let* enabled_if = Lib_info.enabled info in
```

Comment:
refactoring suggestion:

- rename `Lib_info.enabled` to `Lib_info.to_enabled_if`
- add `Lib_info.enabled` which is `val enabled: Lib_info.t -> bool Memo.t`
  - implementation is your conversion from the variant to bool (saw that a few times in the PR)